File: studies/movies/ggplot2/area_colored_stacked.R
library(tidyverse)
library(lubridate)

df <- read_csv("data/movies.csv") |>
  mutate(`Release Date` = year(`Release Date`))
ggplot(df) +
  geom_area(aes(
    x = `Release Date`,
    y = `Worldwide Gross`,
    fill = `MPAA Rating`
  ), stat = "summary", fun = "sum", position = "fill")

File: galleries/movies/altair/area_colored_stacked.py
import pandas as pd
import altair as alt

df = pd.read_csv("data/movies.csv")

chart = (
    alt.Chart(df)
    .mark_area()
    .encode(
        x="year(Release Date)",
        y=alt.Y("sum(Worldwide Gross)").stack("normalize"),
        color="MPAA Rating",
    )
)
